src/stan_math_backend/Transform_Mir.ml

```diff
@@ -2,41 +2,89 @@ open Core_kernel
 open Middle
 
 let opencl_triggers =
-  String.Map.of_alist_exn [("normal_id_glm_propto_lpdf", [0; 1])]
-
+  String.Map.of_alist_exn
+    [ ("normal_id_glm_lpdf",
+        ([0;1],
+          [ (* Array of conditions under which to move to OpenCL *)
+            ([1],[]) (* Argument 1 is data *)
+          ]
+        )
+      )
+    ; ("bernoulli_logit_glm_lpmf",
+        ([0;1], [([1],[])])
+      )
+    ; ("categorical_logit_glm_lpmf",
+        ([0;1], [([1],[(1,UMatrix)])]) (* Additional requirement of argument 1 being a matrix *)
+      )
+    ; ("neg_binomial_2_log_glm_lpmf",
+        ([0;1], [([1],[])])
+      )
+    ; ("ordered_logistic_glm_lpmf",
+        ([0;1], [([1],[(1,UMatrix)])])
+      )
+    ; ("poisson_log_glm_lpmf",
+        ([0;1], [([1],[])])
+      ) 
+    ]
 let opencl_suffix = "_opencl__"
 let to_matrix_cl e = {e with expr= FunApp (StanLib, "to_matrix_cl", [e])}
 
-let rec switch_expr_to_opencl available_cl_vars e =
-  let is_avail = List.mem available_cl_vars ~equal:( = ) in
+let rec switch_expr_to_opencl _available_cl_vars e =
```

Comment:
Why this name change? I think usually the leading underscore means it won't be used, but we use it on the next line right?

Reply:
No reason, will change it back. 

I was trying to understand stuff and I find that for me the best way of learning what things do is commenting out lines and seeing what changes or brakes. And runtest fails if you have unused variables.